New Port: Cute Fame: Halloween Bash (Done on behalf of Fame)

binarycounter opened the pull request:
# New Port for Cute Fame: Halloween Bash

I'm submitting this PR on behalf of Fame, who has trouble setting the repository up.
I've cleaned up a few things: I removed the conf directory, removed a stray log and port.json file hidden within the port folder. I've also corrected the name within port.json from "cutefamehalloweenbash" to "cutefamehalloweenbash.zip" as the PM release builder suggested.

## Game Information
- **Title**: Cute Fame: Halloween Bash
- **URL**: https://cutefame.itch.io/cutefamehalloweenbash

## Submission Requirements

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS
- [X] Knulli

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

TheRealFame commented:
Finallyyyyyyyyyyyyyy

tabreturn commented on ports/cutefamehalloweenbash/README.md:
```diff
@@ -0,0 +1,18 @@
+|A| Special Weapon|
+|B| Main Weapon|
+|X| Swap Weapon|
+|Y| Spwap Special Weapon |
```
|Y| Swap Special Weapon |

tabreturn commented on ports/cutefamehalloweenbash/port.json:
```diff
@@ -0,0 +1,47 @@
+    "version": 4,
+    "name": "cutefamehalloweenbash.zip",
+    "items": [
+        "Cute Fame Halloween Bash"
```
"Cute Fame Halloween Bash.sh",
    "cutefamehalloweenbash"

tabreturn commented on ports/cutefamehalloweenbash/port.json:
```diff
@@ -0,0 +1,47 @@
+        ],
+        "store": [
+            {
+                "name": "Cute Fame: Halloween Bash (Website)",
```
"name": "itch.io",

tabreturn reviewed (changes requested):
In addition to my per-file comments:

1. Is there a license for the game itself? It's ready-to-run (?), but I only see g4les and renpy license files.

2. Rename .sh file from `CuteFameHalloweenBash.sh` to `Cute Fame Halloween Bash.sh`

3. Is it necessary for the port.json `"desc"` value to begin with a new line? And why does the `"desc_md"` contain different wording (it should just be the same as `"desc"` but with additional markdown formatting?). Maybe "Spooktober game jam! 2023" is missing from the beginning of the `"desc"` value?

4. port.json `"inst"` reads `"(Should be plug and play)"` -- is it ready to run or is it not?

5. The port.json contains `"status"` and `"files"` elements -- I've never seen this before? But maybe it's something new / something I don't know about ...

binarycounter commented:
> In addition to my per-file comments:
> 
> 1. Is there a license for the game itself? It's ready-to-run (?), but I only see g4les and renpy license files.

The person who asked me to submit this PR is the author of this game. Since I'm not seeing an open source license on Fame's itch page, I'm assuming "All Rights Reserved". @TheRealFame Please clarify.

> 2. Rename .sh file from `CuteFameHalloweenBash.sh` to `Cute Fame Halloween Bash.sh`

Fixed.

> 3. Is it necessary for the port.json `"desc"` value to begin with a new line? 

Fixed

> And why does the `"desc_md"` contain different wording (it should just be the same as `"desc"` but with additional markdown formatting?). Maybe "Spooktober game jam! 2023" is missing from the beginning of the `"desc"` value?

@TheRealFame Please clarify what you want here.

> 4. port.json `"inst"` reads `"(Should be plug and play)"` -- is it ready to run or is it not?

Changed to "Ready to run!"

> 5. The port.json contains `"status"` and `"files"` elements -- I've never seen this before? But maybe it's something new / something I don't know about ...

I believe this is something that auto-generates when PM creates a placeholder port.json when you autoinstall a port without metadata. I've now used a port.json generated by the tool on the website as the template, should be compliant now.

tabreturn commented:
> > > In addition to my per-file comments:
> > > 
> > > 1. Is there a license for the game itself? It's ready-to-run (?), but I only see g4les and renpy license files.
> > 
> > 
> > The person who asked me to submit this PR is the author of this game. Since I'm not seeing an open source license on Fame's itch page, I'm assuming "All Rights Reserved". @TheRealFame Please clarify.
> > > 2. Rename .sh file from `CuteFameHalloweenBash.sh` to `Cute Fame Halloween Bash.sh`
> > 
> > 
> > Fixed.
> > > 3. Is it necessary for the port.json `"desc"` value to begin with a new line?
> > 
> > 
> > Fixed
> > > And why does the `"desc_md"` contain different wording (it should just be the same as `"desc"` but with additional markdown formatting?). Maybe "Spooktober game jam! 2023" is missing from the beginning of the `"desc"` value?
> > 
> > 
> > @TheRealFame Please clarify what you want here.
> > > 4. port.json `"inst"` reads `"(Should be plug and play)"` -- is it ready to run or is it not?
> > 
> > 
> > Changed to "Ready to run!"
> > > 5. The port.json contains `"status"` and `"files"` elements -- I've never seen this before? But maybe it's something new / something I don't know about ...
> > 
> > 
> > I believe this is something that auto-generates when PM creates a placeholder port.json when you autoinstall a port without metadata. I've now used a port.json generated by the tool on the website as the template, should be compliant now.
> 
> I mainly wanted the game to reserve it's art licenses, Aka most of them being tied to "Pawfulltrust" the artist that helped me make the art, And as I did everything else besides sound design I too would reserve those rights and placehold the musician's rights.
> 
> So whatever license would fit there would most likely be for the contents of the game rather then anything associated with Renpy or it's runtime..
> 
> But yes, The port is supposed to come with the game, And be ready to run!

@TheRealFame -- you'll need to provide a license, please. It doesn't need to be something in official legalese or anything -- something that you've written that grants PortMaster permission to distribute the files and any limitations. A paragraph or two like you've written above is fine, but you need to be a bit more definitive -- like, no "I would" or "whatever license", etc. wording, please.

tabreturn reviewed (commented):
Firstly, thanks for adding the license.

Secondly -- I'm not familiar with Ren'Py, but ...

1. This is using the Ren'Py runtime, right? So you can delete the `## Compile` section in the README, I would assume?
2. I also noticed several `pyc` files scattered among the game files -- are those necessary?
3. Should there be a `saves/persistent` file distributed with the port? I'm assuming this might be a save file (so it shouldn't be present on first run)?

TheRealFame commented:
> Firstly, thanks for adding the license.
> 
> Secondly -- I'm not familiar with Ren'Py, but ...
> 
> 1. This is using the Ren'Py runtime, right? So you can delete the `## Compile` section in the README, I would assume?
> 2. I also noticed several `pyc` files scattered among the game files -- are those necessary?
> 3. Should there be a `saves/persistent` file distributed with the port? I'm assuming this might be a save file (so it shouldn't be present on first run)?

I'll have to ask someone with/using the Renpy runtime about most of these..

tabreturn commented:
> > Firstly, thanks for adding the license.
> > Secondly -- I'm not familiar with Ren'Py, but ...
> > 
> > 1. This is using the Ren'Py runtime, right? So you can delete the `## Compile` section in the README, I would assume?
> > 2. I also noticed several `pyc` files scattered among the game files -- are those necessary?
> > 3. Should there be a `saves/persistent` file distributed with the port? I'm assuming this might be a save file (so it shouldn't be present on first run)?
> 
> I'll have to ask someone with/using the Renpy runtime about most of these..

- I would just delete `## Compile` section in the README.
- Delete the `saves/persistent` file and test the game to see if it starts and saves okay (file should reappear).
- Then, if the above works fine, delete the `pyc` files and see if the game runs (again, these files will likely reappear).

Cebion commented:
> Are requests like this not fixable by someone requesting changes btw? I never had pulled in a change and not modified it afterwards so I wouldn't know the process for files through github instead of code and deletions

The Porter is responsible for submitting and fixing their own pull requests.

binarycounter commented:
Not sure what happened here, i just renamed the branch.... Activity has kind of stopped here. If there's interest in reopening this, I'll remake the PR.

TheRealFame commented:
> Not sure what happened here, i just renamed the branch.... Activity has kind of stopped here. If there's interest in reopening this, I'll remake the PR.

Please do.
